src/parser/symtab.ts

```diff
@@ -0,0 +1,1738 @@
+                        this.cur.symbols !== this.global &&
+                        annAssign.simple
+                    ) {
+                        throw new pySyntaxError(
+                            cur & SYMTAB_CONSTS.DEF_GLOBAL
+                                ? `annotated name '${eName.id}' can't be global`
+                                : `annotated name '${eName.id}' can't be nonlocal`,
+                            [this.filename, s.lineno, s.col_offset, ""]
+                        );
+                    }
+
+                    if (annAssign.simple) {
+                        this.addDef(eName.id, SYMTAB_CONSTS.DEF_ANNOT | SYMTAB_CONSTS.DEF_LOCAL);
+                    } else if (annAssign.value) {
+                        this.addDef(eName.id, SYMTAB_CONSTS.DEF_LOCAL);
+                    }
+                } else {
+                    this.visitExpr(annAssign.target);
+                }
+
+                this.visitExpr(annAssign.annotation);
+                if (annAssign.value) {
+                    this.visitExpr(annAssign.value);
+                }
+                break;
+            }
+            case ASTKind.AugAssign: {
+                const augAssign = s as AugAssign;
+
+                this.visitExpr(augAssign.target);
+                this.visitExpr(augAssign.value);
+                break;
+            }
+            case ASTKind.For: {
+                const for_ = s as For;
+
+                this.visitExpr(for_.target);
+                this.visitExpr(for_.iter);
+                this.SEQ(this.visitStmt, for_.body);
+                if (for_.orelse) this.SEQ(this.visitStmt, for_.orelse);
+                break;
+            }
+            case ASTKind.While: {
+                const while_ = s as While;
+
+                this.visitExpr(while_.test);
+                this.SEQ(this.visitStmt, while_.body);
+                if (while_.orelse) this.SEQ(this.visitStmt, while_.orelse);
+                break;
+            }
+            case ASTKind.If: {
+                const if_ = s as If;
+
+                /* XXX if 0: and lookup_yield() hacks */
+                this.visitExpr(if_.test);
+                this.SEQ(this.visitStmt, if_.body);
+                if (if_.orelse) this.SEQ(this.visitStmt, if_.orelse);
+                break;
+            }
+            case ASTKind.Raise: {
+                const raise = s as Raise;
+
+                if (raise.exc) {
+                    this.visitExpr(raise.exc);
+                    if (raise.cause) {
+                        this.visitExpr(raise.cause);
+                    }
+                }
+                break;
+            }
+            case ASTKind.Try: {
+                const try_ = s as Try;
+
+                this.SEQ(this.visitStmt, try_.body);
+                this.SEQ(this.visitStmt, try_.orelse);
+                this.SEQ(this.visitExcepthandler, try_.handlers as ExceptHandler[]); // @stu this is odd.
```

Comment:
the base class is `excepthandler` rather than `ExceptHandler`
`ExceptHandler` also exists in astnodes but it's not the type of `handlers`.

Reply:
but this is the _only_ place I had to cast one of the properties

Reply:
`visitExceptionhandler` expects an `ExceptionHandler`
but
`Try.handlers` is an `exceptionhandler[]` 
so it could be another subclass of `exceptionhandler` that isn't `ExceptionHandler`.
Just change the `visitExcepthandler(eh: ExceptHandler)` to `visitExcepthandler(eh: exceptionhandler)`
Then you can remove this typecast

Reply:
oh actually that just breaks the code for `visitExcepthandler` I guess we can't win here.
We can address it with the asdl patch #53. Make `Try.handlers: ExceptHandler[]`

Reply:
👍